Modules/_ctypes/stgdict.c

```diff
@@ -411,231 +372,63 @@ PyCStructUnionType_update_stginfo(PyObject *type, PyObject *fields, int isStruct
-            PyErr_Format(PyExc_TypeError,
-                         "second item in _fields_ tuple (index %zd) must be a C type",
-                         i);
-            return -1;
+        if (PyStgInfo_FromType(st, prop->proto, &info) < 0) {
+            goto error;
         }
+        assert(info);
 
         stginfo->ffi_type_pointer.elements[ffi_ofs + i] = &info->ffi_type_pointer;
         if (info->flags & (TYPEFLAG_ISPOINTER | TYPEFLAG_HASPOINTER))
             stginfo->flags |= TYPEFLAG_HASPOINTER;
-        stginfo->flags |= info->flags & (TYPEFLAG_HASUNION | TYPEFLAG_HASBITFIELD);
         info->flags |= DICTFLAG_FINAL; /* mark field type final */
-        if (PyTuple_Size(pair) == 3) { /* bits specified */
-            stginfo->flags |= TYPEFLAG_HASBITFIELD;
-            switch(info->ffi_type_pointer.type) {
-            case FFI_TYPE_UINT8:
-            case FFI_TYPE_UINT16:
-            case FFI_TYPE_UINT32:
-            case FFI_TYPE_SINT64:
-            case FFI_TYPE_UINT64:
-                break;
-
-            case FFI_TYPE_SINT8:
-            case FFI_TYPE_SINT16:
-            case FFI_TYPE_SINT32:
-                if (info->getfunc != _ctypes_get_fielddesc("c")->getfunc
-                    && info->getfunc != _ctypes_get_fielddesc("u")->getfunc)
-                {
-                    break;
-                }
-                _Py_FALLTHROUGH;  /* else fall through */
-            default:
-                PyErr_Format(PyExc_TypeError,
-                             "bit fields not allowed for type %s",
-                             ((PyTypeObject *)desc)->tp_name);
-                Py_DECREF(pair);
-                return -1;
-            }
-            if (bitsize <= 0 || bitsize > info->size * 8) {
-                PyErr_Format(PyExc_ValueError,
-                                "number of bits invalid for bit field %R",
-                                name);
-                Py_DECREF(pair);
-                return -1;
-            }
-        } else
-            bitsize = 0;
-
-        if (isStruct) {
-            const char *fieldfmt = info->format ? info->format : "B";
-            const char *fieldname = PyUnicode_AsUTF8(name);
-            char *ptr;
-            Py_ssize_t len;
-            char *buf;
-            Py_ssize_t last_size = size;
-            Py_ssize_t padding;
-
-            if (fieldname == NULL)
-            {
-                Py_DECREF(pair);
-                return -1;
-            }
 
-            /* construct the field now, as `prop->offset` is `offset` with
-               corrected alignment */
-            prop = PyCField_FromDesc(st, desc, i,
-                                   &field_size, bitsize, &bitofs,
-                                   &size, &offset, &align,
-                                   pack, big_endian, layout_mode);
-            if (prop == NULL) {
-                Py_DECREF(pair);
-                return -1;
-            }
+        assert(prop);
```

Comment:
I don't see how you could have `prop` NULL now.

Reply:
Yeah, that's a leftover :)